source/Calamari/Kubernetes/Integration/CommandLineTool.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Calamari.Common.Features.Processes;
+using Calamari.Common.Plumbing.Logging;
+
+namespace Calamari.Kubernetes.Integration
+{
+    public class CommandLineTool
+    {
+        protected readonly ILog log;
+        protected readonly string workingDirectory;
+        protected readonly Dictionary<string, string> environmentVars;
+
+        readonly ICommandLineRunner commandLineRunner;
+
+        protected CommandLineTool(ILog log, ICommandLineRunner commandLineRunner, string workingDirectory, Dictionary<string, string> environmentVars)
+        {
+            this.log = log;
+            this.commandLineRunner = commandLineRunner;
+            this.workingDirectory = workingDirectory;
+            this.environmentVars = environmentVars;
+        }
+
+        public string ExecutableLocation { get; protected set; }
+
+        protected bool TryExecuteCommandAndLogOutput(string exe, params string[] arguments)
+        {
+            var result = ExecuteCommandAndLogOutput(new CommandLineInvocation(exe, arguments));
+            return result.ExitCode == 0;
+        }
+
+        protected CommandResult ExecuteCommandAndLogOutput(CommandLineInvocation invocation)
```

Comment:
Do we want all these commands along with output logged? I think this is a change of functionality and in the case of some of the `get-credentials` style calls we'd be logging returned tokens?

Reply:
The commands [were always logged by default before, the naming was just never explicit about it](https://github.com/OctopusDeploy/Calamari/pull/960/commits/354732537d5ab491782d1b9078fe7bb63323273c#diff-a201f377c4880673f195e26a67d8323bb7cac40ce3dec8745fcb63a1bb55e335L914). 

Where the commands weren't previously logged, I've introduced the [ExecuteCommandAndReturnOutput](https://github.com/OctopusDeploy/Calamari/pull/960/commits/354732537d5ab491782d1b9078fe7bb63323273c#diff-b42431012ab7a6abafd9cd1104372a0a755df8abdd8292c6e93c775ffb9d581aR79) method, which after the CLI class refactors was only internal calls.

Reply:
The [assent tests](https://github.com/OctopusDeploy/Calamari/pull/960/commits/354732537d5ab491782d1b9078fe7bb63323273c#diff-391bf7573d07d10295da10f256fa5e03290fbce52c977651657598cb2ddf177e) validate that no extra command logging has been introduced unless it was done intentionally :)

Reply:
I think your right that this isn't changing functionality. The assent tests do [swallow a number of command results](https://github.com/OctopusDeploy/Calamari/blob/17f77e2500753f73061a965b27e841fa2a49d32c/source/Calamari.Tests/KubernetesFixtures/KubernetesRunsScriptFixture.cs#L430). I'm going to pull the branch down and give it a run just to step through and understand this properly.

Reply:
Confirmed locally on EKS there are no additional outputs, sorry for the hold up!

Reply:
Thanks for being detailed in your review, I appreciate it :)